src/endpoint/s3/s3_bucket_logging.js

```diff
@@ -0,0 +1,123 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const dbg = require('../../util/debug_module')(__filename);
+const http_utils = require('../../util/http_utils');
+
+function is_bucket_logging_enabled(source_bucket, bucket_name) {
+
+    if (!source_bucket || !source_bucket.bucket_info.logging) {
+        dbg.info("Bucket logging is not configured for bucket : ", bucket_name);
+        return false;
+    }
+
+    const enable = check_logging_config_perm(source_bucket.bucket_info.logging);
+    return enable;
+
+}
+
+function check_logging_config_perm(logging) {
+    // TODO: check if log bucket has permission to upload log objects and other things.
+    return true;
+}
+
+function endpoint_bucket_op_logs(op_name, req, res, source_bucket) {
+
+    dbg.info("Sending op logs for op name = ", op_name);
+    // 1 - Get all the information to be logged in a log message.
+    // 2 - Format it and send it to log bucket/syslog.
+
+    const s3_log = get_bucket_log_record(op_name, source_bucket, req, res);
+    send_bucket_op_logs(s3_log);
+
+}
+
+function get_bucket_log_record(op_name, source_bucket, req, res) {
+
+    const client_ip = http_utils.parse_client_ip(req);
+
+    const log = {
+        op: req.method,
+        bucket_owner: source_bucket.bucket_owner,
+        source_bucket: req.params.bucket,
+        object_key: req.originalUrl,
+        log_bucket: source_bucket.bucket_info.logging.log_bucket,
+        remote_ip: client_ip,
+        request_uri: req.originalUrl,
+        http_status: res.statusCode,
+        request_id: req.request_id
+    };
+
+    const s3_log = new Aws_log_entry();
```

Comment:
See my comment in response to Ben. I think we should format it in the (syslog) server side before writing the log object. I think that for now you can just return `log` as the log record and send it as JSON